src/diffusers/schedulers/scheduling_ddim.py

```diff
@@ -255,17 +263,25 @@ def step(
         # - pred_prev_sample -> "x_t-1"
 
         # 1. get previous step value (=t-1)
-        prev_timestep = timestep - self.config.num_train_timesteps // self.num_inference_steps
+        if isinstance(timestep, tuple) and len(timestep) == 2:
```

Comment:
why do we need this? I don't think we should every pass `timestep` as a tuple no? This would break the API :sweat_smile:

Reply:
yeah, this API change would require rewriting a lot of stuff eventually but for now it's compatible with the old integer inputs too. 

I think the previous way of exactly equally spaced jumps is not general enough. In principle, you could select *any* of the original timesteps for DDIM sampling so the exactly equally spaced assumption in this .step() method is too restricting. For example, if you use a number of DDIM steps that doesn't properly divide the number of training steps, you need to support jump sizes that are not always identical (for example with 64 DDIM steps on a model trained with 1000 steps, we'd sometimes need to jump over 15 and sometimes over 16 training steps). The usecase I have for this is progressive distillation from 512-step DDIM on models that have been trained with 1000 steps because progressive distillation normally needs a 2^x number of steps.

What do you think is the best way to accomplish this?

Reply:
yeah that's a good idea, I'll get on that